mllib/src/test/scala/org/apache/spark/mllib/util/TestingUtilsSuite.scala

```diff
@@ -457,4 +460,19 @@ class TestingUtilsSuite extends SparkFunSuite {
     assert(Matrices.sparse(2, 2, Array(0, 1, 2), Array(0, 1), Array(3.1, 3.5)) !~=
       Matrices.dense(0, 0, Array()) relTol 0.01)
   }
+
+  test("SPARK-31400, catalogString distinguish Vectors in ml and mllib") {
+    val schema = StructType(Array[StructField] {
+      StructField("features", new org.apache.spark.mllib.linalg.VectorUDT)
+    })
+    val e = intercept[IllegalArgumentException] {
+      SchemaUtils.checkColumnType(schema, "features", new VectorUDT)
+    }
+    assert(e.getMessage.contains("must be of type"), "class is not same")
```

Comment:
If you're going to write a test for it, I'd test for the presence of the class name, which is the point of the change

Reply:
I have already written a test and passed  it in my local environment,Could you please test the presence of the class name.

Reply:
? I'm suggesting that the test should test this, if you're bothering to add it. The point of your change is to express the required and actual type in the error message, so the test should check whether at least one of those class names is present.

Reply:
Yes, you are right，so sorry I am not test it right before.

Reply:
I hope this will be better： `assert(e.getMessage.contains(
      "org.apache.spark.mllib.linalg.VectorUDT:struct<type:tinyint,size:int,indices:array<int>"),
      "dataType is not desired")
`